### PR TITLE
[DOCS] Deleting an index doesn't delete Kibana components

### PR DESCRIPTION
**Pull request by stefnestor**
Changes:
* Notes the delete index API can delete multiple indices at once.
* Notes deleting an index deletes its docs, shards, and metadata but does not delete any related Kibana components.
* Relocates a note about deleting a data stream's write index to the description.
* Corrects the default `expand_wildcards` value.

Relates to this [Elastic Discuss topic](https://discuss.elastic.co/t/delete-all-data-from-index-without-deleting-index/87661).

### Preview
https://elasticsearch_77286.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/indices-delete-index.html

### CODE REVIEW

**Comment by jrodewig** [comment]
Thanks for this PR @stefnestor.

Overall, this looks good. However, I don't know if we want to include a wildcard example here. We do document our wildcard support under the `<index>` parameter. However, in 8.0, wildcard expansion will be off by default.

I pushed some changes to reorganize this page a bit and correct an error I spotted. I'll reach out to an engineer to get their review before merging this in.

**Comment by jrodewig** [comment]
@joegallo  Do you mind giving this a look at your convenience?

In particular, I'd like a double-check on the correction to `expand_wildcards`. I tested and checked https://github.com/elastic/elasticsearch/blob/a92a647b9f17d1bddf5c707490a19482c273eda3/server/src/main/java/org/elasticsearch/action/admin/indices/delete/DeleteIndexRequest.java#L30, but I always have a tough time parsing `indicesOptions`. Thanks!

**Review by jrodewig — APPROVED** [review]

**Comment by joegallo** [comment]
@jrodewig that looks precisely correct to me. FWIW, I found that hard to parse, too, I ended up adding a little extra logging code and then just running a bunch of DELETE requests through a local build to confirm the results from the different options:

```diff
diff --git a/server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestDeleteIndexAction.java b/server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestDeleteIndexAction.java
index be24a401e0e..68249a1bf40 100644
--- a/server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestDeleteIndexAction.java
+++ b/server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestDeleteIndexAction.java
@@ -8,6 +8,8 @@

 package org.elasticsearch.rest.action.admin.indices;

+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import org.elasticsearch.action.admin.indices.delete.DeleteIndexRequest;
 import org.elasticsearch.action.support.IndicesOptions;
 import org.elasticsearch.client.node.NodeClient;
@@ -23,6 +25,8 @@ import static org.elasticsearch.rest.RestRequest.Method.DELETE;

 public class RestDeleteIndexAction extends BaseRestHandler {

+    private static final Logger logger = LogManager.getLogger(RestDeleteIndexAction.class);
+
     @Override
     public List<Route> routes() {
         return List.of(
@@ -41,6 +45,9 @@ public class RestDeleteIndexAction extends BaseRestHandler {
         deleteIndexRequest.timeout(request.paramAsTime("timeout", deleteIndexRequest.timeout()));
         deleteIndexRequest.masterNodeTimeout(request.paramAsTime("master_timeout", deleteIndexRequest.masterNodeTimeout()));
         deleteIndexRequest.indicesOptions(IndicesOptions.fromRequest(request, deleteIndexRequest.indicesOptions()));
+
+        logger.info("indicesOptions were " + deleteIndexRequest.indicesOptions());
+
         return channel -> client.admin().indices().delete(deleteIndexRequest, new RestToXContentListener<>(channel));
     }
 }
```

**Review by joegallo — APPROVED** [review]

**Comment by jrodewig** [comment]
Thanks @joegallo! Glad it wasn't just me. 😄 

I'll give your testing approach a try next time I need to test one of these parameters out. Thanks for sharing!
